### DISCUSSION TITLE
Sharepoint breaks things?

### DISCUSSION THREAD

**Discussion by sweharris** [abraunegg/onedrive#3046] [Q&A] [answered]
In a recent issue report I saw

```
CAUTION: When you upload files to Microsoft OneDrive that uses SharePoint as its backend, Microsoft OneDrive will alter your files post upload.
```

Maybe I'm missing something, but I don't see anything in the docs about this; a `grep -ri sharepoint . | grep -i alter` only shows it in `src/sync.d`.

This may become more worrysome because a lot of personal OneDrive accounts are being migrated to the same backend as Business OneDrive (I'm guessing Microsoft is trying to rationalise their infrastructure) and so is probably Sharepoint (I've seen a file with `_vti_` in it get rejected after my drive got migrated, which is a Sharepoint issue).

What is Microsoft doing that could break my files?

**Comment by abraunegg** [comment, marked as answer]
Microsoft via a feature you cannot disable - will willfully modify your file post upload

Please read: https://github.com/OneDrive/onedrive-api-docs/issues/935

This breaks ALL notion of file integrity.

**Reply by sweharris** [reply]
Well, that sucks.  This kinda removes the whole reason for using OneDrive.  What are Microsoft thinking?!
